cpp/include/cuvs/neighbors/vamana.h

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cuvs/core/c_api.h>
+#include <cuvs/distance/distance.h>
+#include <cuvs/neighbors/common.h>
+#include <dlpack/dlpack.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup vamana_c_index_params C API for Vamana index build
+ * @{
+ */
+
+/**
+ * @brief Supplemental parameters to build Vamana Index
+ *
+ * `graph_degree`: Maximum degree of graph; corresponds to the R parameter of
+ * Vamana algorithm in the literature.
+ * `visited_size`: Maximum number of visited nodes per search during Vamana algorithm.
+ * Loosely corresponds to the L parameter in the literature.
+ * `vamana_iters`: The number of times all vectors are inserted into the graph. If > 1,
+ * all vectors are re-inserted to improve graph quality.
+ * `max_fraction`: The maximum batch size is this fraction of the total dataset size. Larger
+ * gives faster build but lower graph quality.
+ * `alpha`: Used to determine how aggressive the pruning will be.
+ */
+struct cuvsVamanaIndexParams {
+  /** Distance type. */
+  cuvsDistanceType metric;
+  /** Maximum degree of output graph corresponds to the R parameter in the original Vamana
+   * literature. */
+  uint32_t graph_degree;
+  /** Maximum number of visited nodes per search corresponds to the L parameter in the Vamana
+   * literature **/
+  uint32_t visited_size;
+  /** Number of Vamana vector insertion iterations (each iteration inserts all vectors). */
+  uint32_t vamana_iters;
```

Comment:
`vamana_iters` is a float value now. Was changed a few PRs ago to better be able to control recall.